extensions/wikia/AdEngine/js/video/uapVideo.js

```diff
@@ -29,17 +39,16 @@ define('ext.wikia.adEngine.video.uapVideo', [
 
 		return porvata.inject(params)
 			.then(function (video) {
-				videoInterface.setup(video, [
-					'progressBar',
-					'pauseOverlay',
-					'volumeControl',
-					'closeButton',
-					'toggleAnimation'
-				], {
+				var uiElements = params.autoPlay ? UITemplate.autoPlay : UITemplate.default;
+
+				log(['VUAP UI elements', uiElements], log.levels.debug, logGroup);
+
+				videoInterface.setup(video, uiElements, {
```

Comment:
IMO, moving setupUI to porvata player isn't the best idea, because porvata is and should stay standalone player, so UI should be not defined inside. 

Maybe let's try @mleczey's suggestion and hide players configuration behind some abstraction?